File: sdk-python/examples/contacts/list_activity.py
"""Run with: python examples/contacts/list_activity.py

Fetch the activity feed (checkouts, refunds, scans, emails, invoice payments)
for a single contact.
"""

from __future__ import annotations

from threecommon import ThreeCommon
from threecommon.contacts import ActivityListParams


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        result = client.contacts.list_activity(
            "cnt_replace_with_real_id",
            ActivityListParams(page_size=20),
        )
        print(f"got {len(result.data)} activity records")
        for event in result.data:
            print(f"  {event.created_at} — {event.type}")


if __name__ == "__main__":
    main()

File: sdk-node/examples/contacts/list-activity.ts
/**
 * Fetch the activity feed (checkouts, refunds, scans, emails, invoice
 * payments) for a single contact.
 *
 * Run:
 *   npx tsx examples/contacts/list-activity.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const result = await client.contacts.listActivity('cnt_replace_with_real_id', {
  pageSize: 20,
})

console.log(`got ${String(result.data.length)} activity records`)
for (const event of result.data) {
  console.log(`  ${event.createdAt} — ${event.type}`)
}
